checks/libs/wmi/sampler.py

```diff
@@ -47,18 +47,57 @@ def get(self, key):
         return super(CaseInsensitiveDict, self).get(key.lower())
 
 
+class ProviderArchitectureMeta(type):
+    """
+    Metaclass for ProviderArchitecture.
+    """
+    def __contains__(cls, provider):
+        """
+        Support `Enum` style `contains`.
+        """
+        return provider in cls._AVAILABLE_PROVIDER_ARCHITECTURES
+
+
+class ProviderArchitecture(object):
+    """
+    Enumerate WMI Provider Architectures.
+    """
+    __metaclass__ = ProviderArchitectureMeta
+
+    # Available Provider Architecture(s)
+    DEFAULT = 0
+    _32BIT = 32
+    _64BIT = 64
+    _AVAILABLE_PROVIDER_ARCHITECTURES = frozenset([DEFAULT, _32BIT, _64BIT])
+
+
 class WMISampler(object):
     """
     WMI Sampler.
     """
+    # Properties
+    _provider = None
+    _formatted_filters = None
+
+    # Type resolution state
+    _property_counter_types = None
+
+    # Samples
+    _current_sample = None
+    _previous_sample = None
+
+    # Sampling state
+    _sampling = False
```

Comment:
Ok thanks, I got confused here.

Might just be me but I'd rather have all the instance attributes initialized in `__init__` though, because potentially having two different values for `Foo.bar` and `m_foo1.bar` confuses me a lot :/